internal/controller/promotions/promotions.go

```diff
@@ -665,3 +662,27 @@ func parseCreateActorAnnotation(promo *kargoapi.Promotion) string {
 	}
 	return creator
 }
+
+func calculateRequeueInterval(p *kargoapi.Promotion) time.Duration {
+	var (
+		defaultRequeueInterval = 5 * time.Minute
+		step                   = p.Spec.Steps[p.Status.CurrentStep]
+		runner                 = promotion.GetStepRunner(step.As)
```

Comment:
As is a user-assigned name for the "instance" of the step. Uses denotes the "kind" of the step, and that's what we should use for the lookup.

```suggestion
		runner                 = promotion.GetStepRunner(step.Uses)
```